app/controlplane/integrations/dependencytrack/cyclonedx/v1/integration.go

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	core "github.com/chainloop-dev/chainloop/app/controlplane/integrations"
+	"github.com/chainloop-dev/chainloop/app/controlplane/integrations/dependencytrack/cyclonedx/v1/uploader"
+	pb "github.com/chainloop-dev/chainloop/app/controlplane/integrations/gen/dependencytrack/cyclonedx/v1"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const Kind = "Dependency-Track"
+
+var _ core.Registrable = (*Integration)(nil)
+var _ core.Attachable = (*Integration)(nil)
+
+type Integration struct {
+	ID string
+}
+
+// Attach attaches the integration service to the given grpc server.
+// In the future this will be a plugin entrypoint
+func NewIntegration() (*Integration, error) {
+	return &Integration{
+		ID: Kind,
+	}, nil
+}
+
+func (i *Integration) PreRegister(ctx context.Context, registrationRequest *anypb.Any) (*core.PreRegistration, error) {
```

Comment:
this is the actual integration with support for both registration and attachment